developer-docs-site/docs/tutorials/full-node/network-identity-fullnode.md

```diff
@@ -0,0 +1,174 @@
+
+1. First, build the `aptos-operational-tool` using the [aptos-labs/aptos-core](https://github.com/aptos-labs/aptos-core) repo.
+
+    - Fork and clone the [aptos-labs/aptos-core](https://github.com/aptos-labs/aptos-core) repo. For example:
+
+    ```
+    $ git clone https://github.com/<YOUR-GITHUB-USERID>/aptos-core.git
+    $ cd aptos-core
+    $ ./scripts/dev_setup.sh
+    $ source ~/.cargo/env
+    ```
+
+    **Using Docker**
+
+    Alternatively, if you are on either macOS or Windows platform, you can use Aptos Docker image. `cd` into the directory for your local public FullNode and start a Docker container with the latest tools, for example:
+
+    ```
+    $ cd ~/my-full-node
+    $ docker run -it aptoslab/tools:devnet /bin/bash
+    ```
+
+2. Run the key generator to produce a hex encoded static x25519 private key. This will be the private key for your network identity.
+
+   ```
+    cargo run -p aptos-operational-tool \
+        -- generate-key \
+        --encoding hex \
+        --key-type x25519 \
+        --key-file /path/to/private-key.txt
+   ```
+
+   **Using Docker**
+
+   Or from inside the `aptoslab/tools` Docker container. Open a new terminal and `cd` into the directory where you started the Docker container for your FullNode. Making sure to provide the full path to where you want the private key TXT file to be stored, run the command as below:
+
+    ```
+    aptos-operational-tool generate-key \
+        --encoding hex \
+        --key-type x25519 \
+        --key-file /path/to/private-key.txt
+    ```
+
+    Example `private-key.txt` file:
+
+    ```
+    $ cat /path/to/private-key.txt
+    B8BD811A91D8E6E0C6DAC991009F189337378760B55F3AD05580235325615C74
+    ```
+
+## Retrieve the public network identity
+
+Run the peer generator on the previous key file:
+
+   ```
+    $ cargo run -p aptos-operational-tool \
+        -- extract-peer-from-file \
+        --encoding hex \
+        --key-file /path/to/private-key.txt \
+        --output-file /path/to/peer-info.yaml
+   ```
+
+   **Using Docker**
+
+   Or from inside the `aptoslab/tools` Docker container:
+
+
+  ```
+  $ aptos-operational-tool extract-peer-from-file \
+      --encoding hex \
+      --key-file /path/to/private-key.txt \
+      --output-file /path/to/peer-info.yaml
+  ```
+
+This will create a YAML file that will have your public identity in it. This is useful if you want to connect your FullNode to a specific upstream FullNode, and that FullNode only allows known identities to connect to them.
+
+Example output `peer-info.yaml`:
+
+ ```
+  ---
+  ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575:
+    addresses: []
+    keys:
+      - ca3579457555c80fc7bb39964eb298c414fd60f81a2f8eedb0244ec07a26e575
+    role: Downstream
```

Comment:
This might be worth asking someone more in the know about. When I ran through the old steps, I ended up with something that says Upstream, not Downstream.

Reply:
Same Upstream when I ran it. I am copy-pasting my peer-info.yaml as an example.

Reply:
'cc @gregnazario for what this should be 😄